great_expectations/validator/validator.py

```diff
@@ -360,6 +375,298 @@ def inst_expectation(*args, **kwargs):
+            if key in override_keys
+        }
+
+        auto: Optional[bool] = expectation_kwargs.get("auto")
+        profiler_config: Optional[
+            Union[
+                RuleBasedProfilerConfig,
+                Dict[
+                    str,
+                    Union[
+                        str,
+                        float,
+                        Optional[Dict[str, Any]],
+                        Optional[Dict[str, Dict[str, Any]]],
+                    ],
+                ],
+            ]
+        ] = expectation_kwargs.get("profiler_config")
+        default_profiler_config: Optional[
+            Union[
+                RuleBasedProfilerConfig,
+                Dict[
+                    str,
+                    Union[
+                        str,
+                        float,
+                        Optional[Dict[str, Any]],
+                        Optional[Dict[str, Dict[str, Any]]],
+                    ],
+                ],
+            ]
+        ] = expectation_impl.default_kwarg_values.get("profiler_config")
+        if auto and profiler_config is None and default_profiler_config is None:
+            raise ValueError(
+                "Automatic Expectation argument estimation requires a Rule-Based Profiler to be provided."
+            )
+
+        configuration: ExpectationConfiguration
+
+        if auto:
+            # Save custom Rule-Based Profiler configuration for reconciling it with optionally-specified default
+            # Rule-Based Profiler configuration as an override argument to "BaseRuleBasedProfiler.run()" method.
+            override_profiler_config: Optional[
+                Union[
+                    RuleBasedProfilerConfig,
+                    Dict[
+                        str,
+                        Union[
+                            str,
+                            float,
+                            Optional[Dict[str, Any]],
+                            Optional[Dict[str, Dict[str, Any]]],
+                        ],
+                    ],
+                ]
+            ]
+            if default_profiler_config:
+                override_profiler_config = copy.deepcopy(profiler_config)
+            else:
+                override_profiler_config = None
+
+            """
+            If default Rule-Based Profiler configuration exists, use it as base with custom Rule-Based Profiler
+            configuration as override; otherwise, use custom Rule-Based Profiler configuration with no override.
+            """
+            profiler_config = default_profiler_config or profiler_config
+
+            # Roundtrip through schema validation to remove any illegal fields add/or restore any missing fields.
+            validated_profiler_config: dict = ruleBasedProfilerConfigSchema.load(
+                profiler_config
+            )
+            profiler_config = ruleBasedProfilerConfigSchema.dump(
+                validated_profiler_config
+            )
```

Comment:
Do we have a ticket to abstract that into a schema-level method? I think it could make for a nice refactor.

Reply:
@cdkini Not yet -- we should do it throughout.

Reply:
Will resolve this comment upon writing or identifying a ticket - I can add one in the backlog when I'm on tomorrow (unless you want to cover it).

Reply:
@cdkini I created a ticket in JIRA -- thanks!